api/src/main/java/com/codeforcommunity/dto/imports/ImportBlocksRequest.java

```diff
@@ -0,0 +1,38 @@
+package com.codeforcommunity.dto.imports;
+
+import com.codeforcommunity.dto.ApiDto;
+import com.codeforcommunity.exceptions.HandledException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ImportBlocksRequest extends ApiDto {
+  private List<BlockImport> blocks;
+
+  public ImportBlocksRequest(List<BlockImport> blocks) {
+    this.blocks = blocks;
+  }
+
+  private ImportBlocksRequest() {}
+
+  public List<BlockImport> getBlocks() {
+    return blocks;
+  }
+
+  public void setBlocks(List<BlockImport> blocks) {
+    this.blocks = blocks;
+  }
+
+  @Override
+  public List<String> validateFields(String fieldPrefix) throws HandledException {
+    String newFieldPrefix = fieldPrefix + "importNeighborhoodRequest.";
+
```

Comment:
Not sure what is happening here but I will trust it works.

Reply:
Essentially this is just calling validateFields for each BlockImport in the this.blocks List. If any of those objects have nulls where they shouldn't then they'll show up as importNeighborhoodRequest.blockImport.fieldName.

The weird return value is making use of Java streams which are very useful when trying to apply a function to every item in a list, actually pretty similar to things you might have done in racket in fundies 1

Reply:
I see, thanks for explaining!